src/Eloquent/Traits/HasGraphRelationshipScopes.php

```diff
@@ -94,7 +95,9 @@ public function scopeWithRelationshipExpression(
 
         $query->getModel()->setTable($name);
 
-        return $query->withRecursiveExpression($name, $expression)->from($name);
+        $query->getQuery()->withRecursiveExpression($name, $expression->getQuery())->from($name);
```

Comment:
`withRecursiveExpression` exists on the underlying query (CTE one) not directly on the builder from this package. Updated the code to call `withRecursiveExpression` on the underlying query builder, and return the main query